subprojects/gradle-plugin/src/main/java/org/spongepowered/gradle/vanilla/internal/model/GroupArtifactVersion.java

```diff
@@ -54,8 +64,19 @@ public static GroupArtifactVersion parse(final String notation) {
     @Value.Parameter
     public abstract @Nullable String version();
 
+    @Value.Parameter
+    public abstract @Nullable String classifier();
+
     @Override
     public final String toString() {
-        return this.group() + ':' + this.artifact() + (this.version() == null ? "" : ':' + this.version());
+        return this.group() + ':'
```

Comment:
This is getting complicated enough that you should replace the ternaries with a StringBuilder